core/commands/repo.go

```diff
@@ -264,7 +268,66 @@ daemons are running.
 	},
 	Type: MessageOutput{},
 	Marshalers: oldcmds.MarshalerMap{
-		oldcmds.Text: MessageTextMarshaler,
+		cmds.Text: MessageTextMarshaler,
+	},
+}
+
+var repoRmRootCmd = &cmds.Command{
+	Helptext: cmdkit.HelpText{
+		Tagline: "Unlink the root used by the files API.",
+		ShortDescription: `
+'ipfs repo rm-root' will unlink the root used by the files API ('ipfs
+files' commands) without trying to read the root itself.  The root and
+its children will then be removed by the garbage collector unless
+pinned.  This command can only run when no ipfs daemons are running.
+`,
+	},
+	Run: func(req cmds.Request, res cmds.ResponseEmitter) {
+		configRoot := req.InvocContext().ConfigRoot
+
+		// Can't use a full node as that interferes with the removal
+		// of the files root, so open the repo directly
+		repo, err := fsrepo.Open(configRoot)
+		if err != nil {
+			res.SetError(err, cmdkit.ErrNormal)
+			return
+		}
+
+		// Get the old root and display it to the user so that they can
+		// can do something to prevent from being garbage collected,
+		// such as pin it
+		dsk := core.FilesRootKey()
+		val, err := repo.Datastore().Get(dsk)
+		switch {
+		case err == ds.ErrNotFound || val == nil:
+			cmds.EmitOnce(res, &MessageOutput{"Files API root not found.\n"})
+		default:
+			c, err := cid.Cast(val.([]byte))
+			cidStr := ""
+			if err != nil {
+				cidStr = c.String()
```

Comment:
if the err isnt empty, we probably shouldnt call `.String()` on the nil cid it returned

Reply:
Oops.  That was suppose to be if `err == nil`.